### PR TITLE
feat: add support for retry policies and dead letter config for aws_cloudwatch_event_target resources

### LINKED ISSUE #17240: Support for RetryPolicy and DeadLetterConfig in aws_cloudwatch_event_target
_This issue was originally opened by @heitorlessa as hashicorp/terraform#27568. It was migrated here as a result of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._
    
<hr>
    

### Current Terraform Version

```
0.14.5 - Latest as of now
```

### Use-cases

AWS launched support for setting a custom retry configuration on a per Event Target basis, including DLQ.

CloudFormation example: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-events-rule.html#aws-resource-events-rule--examples

### Attempted Solutions

Using `aws_cloudformation_stack` resource to make use of additional properties not available in Terraform.

```hcl
resource "aws_cloudformation_stack" "stack_eventbridge_target_dlq" {
  name          = "stack_eventbridge_target_dlq"
  parameters = {
    DlqQueue = var.event_bridge_target_x_dlq_arn
  }
  template_body = <<STACK
{
  "Parameters" : {
    "DlqQueue" : {
      "Type" : "String",
      "Description" : "SQS DLQ Queue for sending events that couldn't reach their targets"
    }
  },
  "RuleWithRetryAndDLQ": {
      "Type": "AWS::Events::Rule",
      "Properties": {
          "Description": "Test Events Rule",
          "Name": "mynewabc",
          "EventPattern": {
            "source": [
                "aws.ec2"
            ]
          },
          "State": "ENABLED",
          "Targets": [
          {
              "Arn": "arn:aws:sqs:us-west-2:081035103721:demoSQS",
              "Id": "Id1234",
              "RetryPolicy": {
                  "MaximumRetryAttempts": 2,
                  "MaximumEventAgeInSeconds": 400
              },
              "DeadLetterConfig": {
                  "Arn": { "Ref": "DlqQueue" }
              }
          }
          ]
      }
  }
}
STACK
}
```

### Proposal

Here's an example using the same properties names in Terraform syntax:

```hcl
resource "aws_cloudwatch_event_target" "ecs_scheduled_task" {
  target_id = "run-scheduled-task-every-hour"
  arn       = aws_ecs_cluster.cluster_name.arn
  rule      = aws_cloudwatch_event_rule.every_hour.name
  role_arn  = aws_iam_role.ecs_events.arn

  # HERE
  retry_policy {
      maximum_retry_attempts = 2,
      maximum_age_in_seconds = 400
  }

  dead_letter_config {
      arn = aws_sqs_dlq_queue.arn
  }
```

### PR DESCRIPTION
**Pull request by sthulb**
Intro
This PR adds support to create Archives in EventBridge.

Release note for CHANGELOG:

resource_aws_cloudwatch_event_target - adds retry policies and dead letter config for aws_cloudwatch_event_target resources

### Tasks

[x] Acceptance test
[x] Update docs

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #17240

Output from acceptance testing:

```
thulsimo@3c22fbef9550 terraform-provider-aws % make testacc TESTARGS='-run=^TestAccAWSCloudWatchEventTarget_retrypolicy_dlc$'
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=^TestAccAWSCloudWatchEventTarget_retrypolicy_dlc -timeout 120m
=== RUN   TestAccAWSCloudWatchEventTarget_retrypolicy_dlc
=== PAUSE TestAccAWSCloudWatchEventTarget_retrypolicy_dlc
=== CONT  TestAccAWSCloudWatchEventTarget_retrypolicy_dlc
--- PASS: TestAccAWSCloudWatchEventTarget_retrypolicy_dlc (80.64s)
PASS
ok      github.com/terraform-providers/terraform-provider-aws/aws       84.015s


...
```

### CODE REVIEW

File: aws/resource_aws_cloudwatch_event_target.go [hunk 1]
```diff
@@ -248,6 +248,40 @@ func resourceAwsCloudWatchEventTarget() *schema.Resource {
 					},
 				},
 			},
+
+			"retry_policy": {
```
Comment (bill-rich) [inline aws/resource_aws_cloudwatch_event_target.go:252]: These should also be set on `Read`.

File: aws/resource_aws_cloudwatch_event_target_test.go [hunk 2]
```diff
@@ -730,6 +765,89 @@ resource "aws_sns_topic" "test" {
 `, ruleName, snsTopicName)
 }
 
+func testAccAWSCloudWatchEventTargetConfig_retrypolicy_dlc(ruleName, targetName, rName string) string {
```
Comment (bill-rich) [inline aws/resource_aws_cloudwatch_event_target_test.go:768]: This should use camal case for everything past the `_`. eg `testAccAWSCloudWatchEventTargetConfig_retryPolicyDlc`

**Review by bill-rich — CHANGES_REQUESTED** [review]
This looks good. I just had a couple requests. Also please add a `.changelog` file.

**Comment by sthulb** [comment]
All items have been resolved – tests pass despite "failing" terrafmt (due to an error outside of my PR)

**Review by bill-rich — APPROVED** [review]

**Pull request merged**